lib/events/pgevents/pgevents.go

```diff
@@ -206,6 +237,13 @@ func New(ctx context.Context, cfg Config) (*Log, error) {
 		return nil, trace.Wrap(err)
 	}
 
+	isCockroach, err := detectCockroach(ctx, pool)
+	if err != nil {
+		cfg.Log.WithError(err).Warn("Failed to detect if using CockroachDB, continuing assuming it's not.")
+	}
```

Comment:
Can we do this check from `pgcommon.SetupAndMigrate` instead, where we're actually in the middle of the transaction that's about to apply the schema and the check becomes infallible? A temporary failure on the first connection opening will result in a permanent failure to set up the audit log backend and a failure to start the auth.